x-pack/plugins/canvas/public/components/page_manager/index.js

```diff
@@ -25,10 +26,19 @@ const mapStateToProps = state => {
 };
 
 const mapDispatchToProps = dispatch => ({
-  addPage: () => dispatch(pageActions.addPage()),
+  addPage: () => {
+    dispatch(pageActions.addPage());
+    dispatch(selectToplevelNodes([]));
+  },
   movePage: (id, position) => dispatch(pageActions.movePage(id, position)),
-  duplicatePage: id => dispatch(pageActions.duplicatePage(id)),
-  removePage: id => dispatch(pageActions.removePage(id)),
+  duplicatePage: id => {
+    dispatch(pageActions.duplicatePage(id));
+    dispatch(selectToplevelNodes([]));
+  },
+  removePage: id => {
+    dispatch(pageActions.removePage(id));
+    dispatch(selectToplevelNodes([]));
```

Comment:
Minor: If the page being removed is not the one that's currently the interactive one, then the selections are still reset.

Reply:
Good catch, thanks!